File: API/DTOs/UserParams.cs
namespace API.DTOs
{
    public class UserParams : PaginationParams
    {
        public string CurrentUserName { get; set; }
        public string Gender { get; set; }  
        public int minAge { get; set; } = 18;
        public int maxAge { get; set; } = 200;
        public string OrderBy { get; set; } = "LastActive";
    }
}

File: client/src/app/_models/userParams.ts
import { PaginationParams } from "./paginationParams";
import { User } from "./user";

export class UserParams extends PaginationParams{
    gender: string;
    minAge: number = 18;
    maxAge: number = 99;
    orderBy: string = 'lastActive';
    constructor(user :User){
        super();
        this.gender = user.gender === "male"? "female":"male";
    }
}